.gitignore

```diff
@@ -13,9 +12,9 @@ build/
 build-*/
 build_*/
 .project
-*.vcxproj.user
 tests/*-output/
 .idea/
+/CMakeSettings.json
```

Comment:
This is an interesting one.  I initially thought about adding this a while ago.  But then I decided against it as the contents of the file are not specific to the location on a drive, and so having an initial set of builds committed may be useful.

I am going to revisit what the issues are with CMake because I would like to get to one build system and the other changes to handle these are painful.  (I would consider dropping the regression tests from the Visual Studio project and just documenting how to run them from the command line).

Reply:
Would you like me to revert this change?

Reply:
Yes please.  Sorry for not being clear above.

Reply:
Done